samza-kv-rocksdb/src/main/java/org/apache/samza/storage/kv/RocksDbTableProvider.java

```diff
@@ -65,6 +64,38 @@ public RocksDbTableProvider(TableSpec tableSpec) {
     // Enable host affinity
     tableConfig.put(ClusterManagerConfig.CLUSTER_MANAGER_HOST_AFFINITY_ENABLED, "true");
 
+    // Adjust changelog topic setting, when
+    //
+    //  1) TTL is set for RocksDB and
+    //  2) Changelog is enabled
+    //  3) User has not set them
+    //
+    // And we would
+    //  1) Disable log compaction on Kafka changelog topic
+    //  2) Set topic TTL to be slightly larger than RocksDB TTL
+    //
+    String rocksDbTtl = tableSpec.getConfig().get("rocksdb." + RocksDbTableDescriptor.ROCKSDB_TTL_MS);
+    String changelogEnabled = tableSpec.getConfig().get(BaseLocalStoreBackedTableDescriptor.INTERNAL_ENABLE_CHANGELOG);
+    if (rocksDbTtl != null && "true".equalsIgnoreCase(changelogEnabled)) {
+
+      String changelogPrefix = String.format(StorageConfig.CHANGELOG_STREAM(), tableSpec.getId()) + ".";
+
+      // We should use regular Kafka topic instead of log-compacted
```

Comment:
Could use both compact and delete.

Reply:
Why add kafka-specific settings to the RocksDbTableProvider?

Reply:
If we do this this shouldn't be only for the table API. We should have the same behavior for regular stores as well.

Reply:
Moved this to KafkaConfig, logic is still the same. User could override it own config if desired.